src/main/java/com/fasterxml/jackson/databind/ser/BeanPropertyWriter.java

```diff
@@ -350,8 +355,12 @@ protected BeanPropertyWriter(BeanPropertyWriter base, SerializedString name) {
                     base._internalSettings);
         }
         _cfgSerializationType = base._cfgSerializationType;
-        _dynamicSerializers = base._dynamicSerializers;
-        _suppressNulls = base._suppressNulls;
+        if (base.needToResetSerialization()) {
+            _dynamicSerializers = PropertySerializerMap.emptyForProperties();
+        } else {
+            _dynamicSerializers = base._dynamicSerializers;
+        }
+		_suppressNulls = base._suppressNulls;
```

Comment:
Where are these constructors called? Does this mean serializer information from 'base' should be lost?

Reply:
`_dynamicSerializers` is purely performance optimization, caching to avoid unnecessary full introspection -- should never be needed for correctness. Hence if necessary we can just use empty instance for copy constructors.